### ISSUE TITLE
Improve Error Handling

### ISSUE THREAD

**Issue by brimigs** [otter-sec/anchor#2898]
I was getting the following error: `TypeError=: src .toArrayLike is not a function` when the issue with my code was just that the front end was passing through the parameters of the anchor instruction in the incorrect order. 

The frontend was passing addEmployee(name, salary) while the anchor program was expecting addEmployee(salary, name). 

Is there a way to improve this error?

**Comment by acheroncrypto** [comment]
> Is there a way to improve this error?

Yeah, surely! There is a lot of room for improvement with regards to error handling.

**Comment by Jenifer-TheCoder** [comment]
Hi , I want to start contributing to this project. can you assign to me?

**Comment by abhilov23** [comment]
Hey @brimigs , I want to contribute to this issue.

**Comment by Aursen** [comment]
> Hey @brimigs , I want to contribute to this issue. 

Feel free to open a PR to solve the issue

**Comment by xeodus** [comment]
Hey @brimigs, it looks like the issue is still open, I would like to work on it.
